neo4j/session_with_context_test.go

```diff
@@ -181,6 +181,26 @@ func TestSession(outer *testing.T) {
 			assertCleanSessionState(t, sess)
 		})
 
+		inner.Run("Fails fast on context timeout", func(t *testing.T) {
+			_, pool, sess := createSession()
+			sess.driverConfig.MaxTransactionRetryTime = 10 * time.Second
+			pool.BorrowConn = &ConnFake{Alive: false}
+			numRetries := 0
+			ctx := context.Background()
+			ctx, cancel := context.WithTimeout(ctx, 0)
+			defer cancel()
+			<-ctx.Done()
```

Comment:
This unusual method of cancelling the context causes it ctx.Err() to return "deadline exceeded", which represents a worst-case scenario - without the change to Continue, it will process the error as retryable and loop until it hits the MaxDeadConnections limit. By comparison, just cancelling the context the usual way will fail after one attempt, because the "context cancelled" error isn't retryable.